modules/xfem/src/base/XFEMGeometricCut3D.C

```diff
@@ -0,0 +1,168 @@
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "XFEMGeometricCut3D.h"
+
+#include "MooseError.h"
+#include "XFEMFuncs.h"
+
+XFEMGeometricCut3D::XFEMGeometricCut3D(Real t0, Real t1):
+  XFEMGeometricCut(t0,t1),
+  _center(Point(0.0,0.0,0.0)),
+  _normal(Point(0.0,0.0,0.0))
+{}
+
+XFEMGeometricCut3D::~XFEMGeometricCut3D()
+{}
+
+bool
+XFEMGeometricCut3D::cutElementByGeometry(const Elem* /*elem*/,
+                                         std::vector<cutEdge> & /*cutEdges*/,
+                                         Real /*time*/)
+{
+  mooseError("invalid method for 3D mesh cutting");
+  return false;
+}
+
+bool
+XFEMGeometricCut3D::cutElementByGeometry(const Elem* elem,
+                                         std::vector<cutFace> & cutFaces,
+                                         Real /*time*/)
+//TODO: Time evolving cuts not yet supported in 3D (hence the lack of use of the time variable)
+{
+  bool cut_elem = false;
+
+   int hex_indices[6][4] = {{0,3,2,1},{0,1,5,4},{1,2,6,5},{2,3,7,6},{3,0,4,7},{4,5,6,7}};
+   int tet_indices[4][3] = {{0,2,1},{0,1,3},{1,2,3},{2,0,3}};
+
+   unsigned int n_face_nodes = 0;
+   if (elem->n_nodes() == 8)
+     n_face_nodes = 4;
+   else if (elem->n_nodes() == 4)
+     n_face_nodes = 3;
+   else
+     mooseError("this method only works for lienar hexes and tets");
+
+    for (unsigned int i=0; i < elem->n_sides(); i++)
+    {
+      std::vector<unsigned int> cut_edges;
+      std::vector<Real> cut_pos;
+
+      for (unsigned int j=0; j < n_face_nodes; j++)
+      {
+        unsigned int jplus1(j < (n_face_nodes-1) ? (j+1) : 0);
+        unsigned int node_id1 = 0;
+        unsigned int node_id2 = 0;
+        if (elem->n_nodes() == 8)  // hex
+        {
+          node_id1 = hex_indices[i][j];
+          node_id2 = hex_indices[i][jplus1];
+        }else if (elem->n_nodes() == 4) // tet
+        {
+          node_id1 = tet_indices[i][j];
+          node_id2 = tet_indices[i][jplus1];
+        }
+        Node *node1 = elem->get_node(node_id1);
+        Node *node2 = elem->get_node(node_id2);
+        Point p1((*node1)(0), (*node1)(1), (*node1)(2));
+        Point p2((*node2)(0), (*node2)(1), (*node2)(2));
+
+        Point pint(0.0,0.0,0.0);
```

Comment:
again: no need to explicitly call the CTOR with 0,0,0, just use the default constructor!